scripts/deployment/test_model_path_detection.py

```
@@ -0,0 +1,105 @@
+        print(f"  Match: {detected_path == expected_path}")
+        
+        # Clean up environment variable
+        if 'MODEL_BASE_DIR' in os.environ:
+            del os.environ['MODEL_BASE_DIR']
+    
+    # Test 4: With expanduser (~) path
+    print("\n🏠 Test 4: With home directory path expansion")
+    os.environ['SAMO_DL_BASE_DIR'] = "~/Projects/SAMO-DL"
+    
+    detected_path = get_model_base_directory()
+    expected_path = os.path.expanduser("~/Projects/SAMO-DL/deployment/models")
```

Ruff (0.12.2)

`os.path.expanduser()` should be replaced by `Path.expanduser()`

(PTH111)